lightning/src/ln/channelmanager.rs

```diff
@@ -10091,7 +10354,7 @@ where
 		// open_channel message - pre-funded channels are never written so there should be no
 		// change to the contents.
 		let _persistence_guard = PersistenceNotifierGuard::optionally_notify(self, || {
-			let res = self.internal_open_channel(counterparty_node_id, msg);
+			let res = self.internal_open_channel(counterparty_node_id, OpenChannelMessage::V1(msg.clone()));
```

Comment:
Could we have `OpenChannelMessage` hold a reference to avoid this clone? Or make a `OpenChannelMessageRef` type if that is needed to hold values?